sdk/src/types/block/output/unlock_condition/mod.rs

```diff
@@ -498,162 +498,41 @@ mod test {
 }
 
 pub mod dto {
-    use alloc::format;
-
-    use serde::{Deserialize, Serialize, Serializer};
-    use serde_json::Value;
-
-    pub use self::{
-        address::dto::AddressUnlockConditionDto, expiration::dto::ExpirationUnlockConditionDto,
-        governor_address::dto::GovernorAddressUnlockConditionDto,
-        immutable_account_address::dto::ImmutableAccountAddressUnlockConditionDto,
-        state_controller_address::dto::StateControllerAddressUnlockConditionDto,
-        storage_deposit_return::dto::StorageDepositReturnUnlockConditionDto, timelock::dto::TimelockUnlockConditionDto,
-    };
+    use serde::{Deserialize, Serialize};
+
+    pub use self::storage_deposit_return::dto::StorageDepositReturnUnlockConditionDto;
     use super::*;
     use crate::types::{block::Error, TryFromDto};
 
-    #[derive(Clone, Debug, Eq, PartialEq, From)]
+    #[derive(Clone, Debug, Eq, PartialEq, From, Serialize, Deserialize)]
+    #[serde(untagged)]
     pub enum UnlockConditionDto {
         /// An address unlock condition.
-        Address(AddressUnlockConditionDto),
+        Address(AddressUnlockCondition),
         /// A storage deposit return unlock condition.
         StorageDepositReturn(StorageDepositReturnUnlockConditionDto),
```

Comment:
does this have to remain the dto type?

Reply:
Yeah, unfortunately. The reason is that the conversion from `Dto` -> Base type can be validated with token supply.